jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java

```diff
@@ -101,6 +152,69 @@ public static void setCRIURestoreMode() {
 					System.out.println(provider + " restored.");
 				}
 			}
+			// If any algorithms are available from when we took a checkpoint in our algorithm cache
+			// we can now allow callers to get values from this cache.
+			if (!cachedAlgorithms.isEmpty()) {
+				if (debug) {
+					System.out.println("Algorithm cache enabled.");
+				}
+				isCacheEligible = true;
+			}
 		}
 	}
+
+	/**
+	 * Gets algorithms from the checkpointed cache of algorithms.
+	 *
+	 * @param serviceName the name of the Java cryptographic
+	 * service (e.g., Signature, MessageDigest, Cipher, Mac, KeyStore).
+	 * Note: this parameter is case-insensitive.
+	 *
+	 * @return a Set of Strings containing the names of all available
+	 * algorithms or types for the specified Java cryptographic service
+	 * or an empty set if no provider supports the specified service.
+	 */
+	public static Set<String> getAlgorithms(String serviceName) {
+		if ((serviceName == null) || (serviceName.isEmpty())) {
```

Comment:
Please remove the redundant parentheses in `(serviceName.isEmpty())`.

Reply:
Removed redundant parentheses on this line.

Reply:
Put these back in. I wasnt sure if i really really needed these as i was shooting for the bare minimum parens. I figured the conditional before the || would be evaluated ok.